lean4-infoview/src/infoview/rpcSessions.tsx

```diff
@@ -268,6 +279,8 @@ export function WithRpcSessions({ children }: { children: React.ReactNode }) {
         []
     )
 
+    useEvent(ec.events.serverRestarted, () => sessions.ensureAllSessionsClosed())
```

Comment:
Instead of adding even more complexity to `RpcSessions`, why don't we leave it as-is, and instead call `dispose()`, then make a `new RpcSessions(ec)` here?

Reply:
That's not a bad idea!  But I believe the closed flag will still be necessary to prevent references from the old session to be released.

Reply:
Ah yes, unfortunately. But we should still make a new `RpcSessions` object to make sure the promise chain in `connectAt` does not make modifications to `this` after a server restart (I don't think this can happen, but then again I didn't think the bugs we've seen would happen either). In that case `dispose` does not need to clear out `#connected/#connecting`, just has to clear the timeouts and set `#closed` so the `FinalizationRegistry` stops sending `release`s.